src/main/java/Ui.java

```diff
@@ -0,0 +1,156 @@
+ * Deals with user interactions.
+ * Read data input from user.
+ * Print messages to inform/update user according to Duke actions.
+ */
+public class Ui {
+    private final Scanner sc;
+
+    /**
+     * Constructor for Ui class. Initialises a Scanner to read in user input.
+     */
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    /**
+     * Prints the Duke Logo.
+     */
+    public void printDukeLogo() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(logo);
+    }
+
+    /**
+     * Prints a partition.
+     */
+    public void printPartition() {
+        System.out.println("\n/**********************************************************/\n");
+    }
+
+    /**
+     * Prints the introduction.
+     */
+    public void printIntroduction() {
+        printDukeLogo();
+        System.out.println("Hello! I'm Duke.\n" + "What can I do for you?");
+        printPartition();
+    }
+
+    /**
+     * Prints the exit message.
+     */
+    public void printExitMessage() {
+        System.out.println("Bye. Hope to see you again soon!\n");
+        printDukeLogo();
+    }
+
+    /**
+     * Prints the IO error.
+     */
+    public void printStorageIoError() {
+        System.out.println("OOPS! An error occurred while attempting to create/retrieve storage file.");
+        printPartition();
+    }
+
+    /**
+     * Prints the IO error.
+     * @param ie The IO exception.
+     */
+    public void printUpdateIoError(IOException ie) {
+        System.out.println("OOPS! An error occurred while attempting to update storage file:\n"
+                + ie.getMessage());
+        printPartition();
+    }
+
+    /**
+     * Prints the Duke Exception.
+     * @param de The Duke Exception.
+     */
+    public void printDukeException(DukeException de) {
+        System.out.println("OOPS! " + de.getMessage());
+        printPartition();
+    }
+
+    /**
+     * Prints the list of tasks.
+     * @param tasks
+     */
+    public void printTasks(TaskList tasks) {
+        ArrayList<Task> taskList = tasks.getTasks();
+        for (int i = 1; i <= taskList.size(); i++) {
+            System.out.println("" + i + ". " + taskList.get(i - 1).getStatus());
+        }
+        printPartition();
+    }
+
+    /**
+     * Prints the list of tasks according to the input.
+     * @param tasks The list of tasks to print.
+     */
+    public void printSomeTasks(ArrayList<Task> tasks) {
+        for (int i = 1; i <= tasks.size(); i++) {
+            System.out.println("" + i + ". " + tasks.get(i - 1).getStatus());
+        }
+        printPartition();
+    }
+
+    /**
+     * Prints the number of tasks.
+     * @param taskList The TaskList that is being kept tracked of.
+     */
+    public void printNumTasks(TaskList taskList) {
```

Comment:
Perhaps you could change the variable name from taskList to tasks. Same for the other methods